### ISSUE TITLE
Failed CouchDb _all_docs request pending in PouchDb, blocking replication in Node 16

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#7810]
**Describe the bug**
The following `_all_docs` is aborted by CouchDb (2.3.1):
```
[diana@doina ~]$ curl "http://admin:pass@127.0.0.1:6984/medic/_all_docs?keys=[\"e0e46c7bbae044b7eb4507c6badcb70c\",\"ea449934dcd5e506dcede2803a47fc31\"]&include_docs=true&conflicts=true" --globoff -v    
*   Trying 127.0.0.1:6984...
* Connected to 127.0.0.1 (127.0.0.1) port 6984 (#0)
* Server auth using Basic with user 'admin'
> GET /medic/_all_docs?keys=["e0e46c7bbae044b7eb4507c6badcb70c","ea449934dcd5e506dcede2803a47fc31"]&include_docs=true&conflicts=true HTTP/1.1
> Host: 127.0.0.1:6984
> Authorization: Basic YWRtaW46cGFzcw==
> User-Agent: curl/7.85.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Cache-Control: must-revalidate
< Content-Type: application/json
< Date: Wed, 21 Sep 2022 09:56:23 GMT
< Server: CouchDB/2.3.1 (Erlang OTP/20)
< Transfer-Encoding: chunked
< X-Couch-Request-ID: c4e6eb77a0
< X-CouchDB-Body-Time: 0
< 
{"total_rows":913370,"offset":null,"rows":[
* transfer closed with outstanding read data remaining
* Closing connection 0
curl: (18) transfer closed with outstanding read data remaining
```

The request fails because the 2nd document produces a read failure:
```
[emergency] 2022-09-21T09:57:05.570102Z nonode@nohost <0.20174.644> -------- File corruption in <0.16669.2> at position 51824286
[error] 2022-09-21T09:57:05.570235Z nonode@nohost <0.20174.644> -------- rexi_server: from: nonode@nohost(<0.27025.642>) mfa: fabric_rpc:open_doc/3 exit:{file_corruption,<<"file corruption">>} [{couch_file,pread_iolist,2,[{file,"src/couch_file.erl"},{line,184}]},{couch_file,pread_binary,2,[{file,"src/couch_file.erl"},{line,169}]},{couch_file,pread_term,2,[{file,"src/couch_file.erl"},{line,157}]},{couch_bt_engine,read_doc_body,2,[{file,"src/couch_bt_engine.erl"},{line,343}]},{couch_db,make_doc,5,[{file,"src/couch_db.erl"},{line,1699}]},{couch_db,open_doc_int,3,[{file,"src/couch_db.erl"},{line,1637}]},{couch_db,open_doc,3,[{file,"src/couch_db.erl"},{line,281}]},{fabric_rpc,with_db,3,[{file,"src/fabric_rpc.erl"},{line,334}]}]
[error] 2022-09-21T09:57:05.570359Z nonode@nohost <0.27025.642> -------- _all_docs open error: medic ea449934dcd5e506dcede2803a47fc31 :: {error,{case_clause,{error,read_failure}}} [{fabric_view_all_docs,open_doc_int,4,[{file,[115,114,99,47,102,97,98,114,105,99,95,118,105,101,119,95,97,108,108,95,100,111,99,115,46,101,114,108]},{line,269}]},{fabric_view_all_docs,open_doc,4,[{file,[115,114,99,47,102,97,98,114,105,99,95,118,105,101,119,95,97,108,108,95,100,111,99,115,46,101,114,108]},{line,258}]}]
[notice] 2022-09-21T09:57:05.570444Z nonode@nohost <0.25226.634> 65e73c8702 127.0.0.1:6984 172.17.0.1 admin GET /medic/_all_docs?keys=["e0e46c7bbae044b7eb4507c6badcb70c","ea449934dcd5e506dcede2803a47fc31"]&include_docs=true&conflicts=true 200 aborted 1
[error] 2022-09-21T09:57:05.570496Z nonode@nohost <0.25226.634> 65e73c8702 Response abnormally terminated: {<<"ea449934dcd5e506dcede2803a47fc31">>,{case_clause,{error,read_failure}}}
```

If I request the same docs with PouchDb http adapter, the request just hangs. 

**To Reproduce**
1. Corrupt a db file (steps to do that in a controlled way is TBD)
2. Login with a user that wants to download a corrupted document
3. Login hangs forever, with no indication that an error was produced

**Expected behavior**
Corrupted database errors should be handled gracefully.

**Environment**
- Version: 4.0

**Additional context**
This seems to be an issue deep in how `node-fetch@2` (which is the peer dependency of PouchDb) handles incomplete requests using Node 16. `node-fetch@3` seems to handle the failed request correctly. 
This points to PouchDb and `node-fetch@2` peer dependency not correctly supporting Node 16.

**Comment by garethbowen** [comment]
@dianabarsan Is this a dupe (or at least fixed) by #8173 ?

**Comment by dianabarsan** [comment]
Yes, based on the additional context I had included, it seems to be. Thanks for flagging this!
